### PR TITLE
Missing header files

### PR DESCRIPTION
**Pull request by paddyroddy**
When building the `cmake` on a recent upgrade of command line tools for mac it errored on missing header files
image

this PR merely imports `so3_adjoint.h` in two files and besides that it is IDE automatic formatting changes

### CODE REVIEW

File: src/main/c/s2let_analysis_adjoint.c [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #include <math.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <so3_adjoint.h>
```
Comment (paddyroddy) [inline src/main/c/s2let_analysis_adjoint.c]: main change: 1

File: src/main/c/s2let_synthesis_adjoint.c [hunk 2]
```diff
@@ -9,6 +9,7 @@
 #include <math.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <so3_adjoint.h>
```
Comment (paddyroddy) [inline src/main/c/s2let_synthesis_adjoint.c]: main change: 2

**Comment by auggiemarignier** [comment]
This cause `make` to fail for me
Screenshot 2020-08-21 at 10 52 50
However changing the ssht inclusions in the header files works

File: include/s2let_alloc.h [hunk 3]
```diff
@@ -5,7 +5,7 @@
 #ifndef S2LET_ALLOC
 #define S2LET_ALLOC
 
-#include "ssht_types.h"
+#include <ssht.h>
```
Comment (auggiemarignier) [inline include/s2let_alloc.h:8]: Making this change in all the `s2let_*.h` files

**Comment by paddyroddy** [comment]
> This cause `make` to fail for me
> Screenshot 2020-08-21 at 10 52 50
> However changing the ssht inclusions in the header files works

Your fix doesn't work for me
image

**Comment by auggiemarignier** [comment]
@paddyroddy I think I've found something. `so3` has 2 `so3.h` files: one in `include/c` and another in `src/c`.  The latter one doesn't include `so3_adjoint.h` but the former does.  I suspect when you ran `cmake` for `s2let` the `So3_INCLUDE_DIR` was `src/c`.  If you retry with `include/c` instead it should work.  If it does then we'll have to fix `so3` to have just the one header file.

Also we must be using slightly different compilers because your error is just a warning for me if I have `src/c` as the `So3_INCLUDE_DIR`

**Comment by paddyroddy** [comment]
@auggiemarignier yep you're right, well spotted

This is what I was running, but changing to `include` for SO3 works
`cmake .. -DSsht_INCLUDE_DIR=$SSHT/src/c/ -DSsht_LIBRARY=$SSHT/build/src/c/libssht.a -DSo3_INCLUDE_DIR=$SO3/src/c/ -DSo3_LIBRARY=$SO3/build/libso3.a -Dtests=OFF && make clean all`

depending on which one we end up with we'll need to adjust `setup.py` too https://github.com/astro-informatics/src_s2let/blob/605fd8764df046e37889e4e3dad52c37334b447c/setup.py#L19

**Comment by auggiemarignier** [comment]
Awesome.  We can make that change when the PR on `so3` has been approved.  In the mean time I'll close this PR

**Pull request closed without merging**